src/objectives.h

```diff
@@ -0,0 +1,24 @@
+#ifndef SGDNET_OBJECTIVES_
+#define SGDNET_OBJECTIVES_
+
+#include <RcppArmadillo.h>
+
+namespace sgdnet {
+  class Objective {
+  public:
+    virtual double Loss(const arma::mat& p, const arma::mat& y) = 0;
+    virtual arma::mat Gradient(const arma::mat& p, const arma::mat& y) = 0;
+  };
+
+  class Gaussian : public Objective {
+  public:
+    double Loss(const arma::mat& p, const arma::mat& y) {
```

Comment:
Can we call this "mu" to match the statistical literature? (mu = conditional mean = E[Y|X])

Reply:
Hm, is mu really used that often for conditional mean? I suggest just spelling out `prediction` or `conditonal_mean` instead.

Reply:
Fine with me -- I'm always happy with more descriptive names. ("p" just has another meaning as number of features) Both names work well, though I'm slightly biased towards "conditional mean" as "prediction" carries a few more connotations.